test/python/pulse/test_schedule.py

```diff
@@ -796,5 +795,61 @@ def test_different_name_equal(self):
                          Schedule((0, ShiftPhase(0, DriveChannel(1), name='fc2')), name='s2'))
 
 
+class TestTimingUtils(QiskitTestCase):
+    """Test the Schedule helper functions."""
+
+    def test_overlaps(self):
+        """Test the `_overlaps` function."""
+        # pylint: disable=invalid-name
+        a = (0, 1)
+        b = (1, 4)
+        c = (2, 3)
+        d = (3, 5)
+        self.assertFalse(_overlaps(a, b))
```

Comment:
It is typically considered bad practice to unit test private methods. It is sometimes necessary, but can also be a code smell.

Reply:
You don't believe that there might be functions that should be marked as private?

Reply:
If they are not facing the public they are an implementation detail and typically should not be directly tested, but rather tested indirectly through the public interface. Doing so can make refactoring much harder. I won't block on this it is just good practice and not an absolute rule, but it is something to consider.